content/ADR-180-communication-adapters.md

```diff
@@ -0,0 +1,125 @@
+
+The basic structure of a connection string is defined with the following BNF
+
+```bnf
+PROTOCOL          := !':' [A-Za-z][A-Za-z0-9-]+
+PARAMETERS        := any+
+CONNECTION_STRING := PROTOCOL ':' PARAMETERS
+```
+
+Here are some examples:
+
+```
+ws-room:ws-room-service.decentraland.org/rooms/123-room-id
+^^^^^^^                                                    protocol
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ parameter
+
+offline:offline
+^^^^^^^         protocol
+        ^^^^^^^ parameter
+
+livekit:wss://livekit.decentral.io?access_token=abc1223123
+^^^^^^^                                                    protocol
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ parameter
+
+```
+
+## Specification
+
+The realms will provide the ways to get a connection string for our session, with that connection string a communications adapter must be created, which will use a communications transport under the hood.
+
+All communications adapters must implement the semantics defined in [ADR-104](/adr/ADR-104). Some adapters, like livekit, may take advantage of their underlying implementations to set up Voice chat.
+
+> TODO: Document the voicechat semantics
+
+### Resolution of communications transport
+
+1. Get the connection string
+1. Parse it and separate the components protocol and params `[protocol, params] = str.split(':', max: 2)`
+1. Find the function that handles the `protocol` and call it with the `params`
+
+### Protocols
+
+Here is the list of officially supported protocols
+
+- `offline` will create an empty adapter. This protocol is used to test without communications servers and by some single-player experiences like the onboarding tutorial. The parameters for the offline protocol are implementation dependent.
+- `simulator` will create different simmulation conditions to evaluate performance and use cases. The parameters for the offline protocol are implementation dependent.
+- `ws-room` will create a WebSocket transport, using the wire protocol and semantics defined in [ADR-105](/adr/ADR-105). The parameter is the absolute URL to be connected to, including the hostname. If no protocol is defined in the url, `wss://` will be prepended.
+- `livekit` will create a LiveKit transport. The parameter is the absolute URL to be connected to, including the hostname and the livekit token if needed. It is mandatory to specify the `wss://` protocol as part of the URL.
+- `signed-login` performs a signed-fetch as defined in [ADR-44](/adr/ADR-44). It is used to authenticate the user BEFORE sending the final connection string. The connection string comes as part of the responsein the `fixedAdapter` JSON field. In case of an error or access denied, a `message` can be sent to the user to provide more information about the reason of the denyed access.
```

Comment:
```suggestion
- `signed-login` performs a signed-fetch as defined in [ADR-44](/adr/ADR-44). It is used to authenticate the user BEFORE sending the final connection string. The connection string comes as part of the response in the `fixedAdapter` JSON field. In case of an error or denied access, a `message` can be sent to the user to provide more information about the reason for the denial.
```